cmd/monaco/main.go

```diff
@@ -44,380 +36,211 @@ func Run(args []string) int {
 	return RunImpl(args, util.NewFileReader())
 }
 
-func containsVersionFlag(args []string) bool {
-	for _, arg := range args {
-		if arg == "--version" || arg == "-version" {
-			return true
-		}
-	}
-
-	return false
-}
-
 func RunImpl(args []string, fileReader util.FileReader) (statusCode int) {
-	if containsVersionFlag(args) {
-		fmt.Println(version.MonitoringAsCode)
-		return 0
-	}
-
-	statusCode = 0
-
-	dryRun, verbose, environments, projectNameToDeploy, path, errorList, flagError := parseInputCommand(args, fileReader)
-
-	if flagError != nil {
-		util.FailOnError(flagError, "could not parse flags")
-	}
+	var app *cli.App
 
-	var deploymentErrors = make(map[string]error)
-
-	for i, err := range errorList {
-		configIssue := fmt.Sprintf("environmentfile-issue-%d", i)
-		deploymentErrors[configIssue] = err
-	}
-
-	err := util.SetupLogging(verbose)
-
-	if err != nil {
-		util.Log.Error("Error writing log file: %s", err.Error())
+	if newCli, ok := os.LookupEnv("NEW_CLI"); ok && newCli != "0" {
+		app = buildExperimentalCli(fileReader)
+	} else {
+		app = buildCli(fileReader)
 	}
 
-	util.Log.Info("Dynatrace Monitoring as Code v" + version.MonitoringAsCode)
-
-	apis := createApis()
+	err := app.Run(args)
 
-	projects, err := project.LoadProjectsToDeploy(projectNameToDeploy, apis, path, fileReader)
 	if err != nil {
-		util.FailOnError(err, "Loading of projects failed")
-	}
-
-	util.Log.Info("Executing projects in this order: ")
-
-	for i, project := range projects {
-		util.Log.Info("\t%d: %s (%d configs)", i+1, project.GetId(), len(project.GetConfigs()))
-	}
-
-	for _, environment := range environments {
-		err := execute(environment, projects, dryRun, path)
-		if err != nil {
-			deploymentErrors[environment.GetId()] = err
-		}
-	}
-
-	for environment, err := range deploymentErrors {
-		if dryRun {
-			util.Log.Error("Validation of %s failed with error %s\n", environment, err)
-		} else {
-			util.Log.Error("Deployment to %s failed with error %s\n", environment, err)
-		}
-		statusCode = -1
-	}
-
-	if statusCode == 0 {
-		if dryRun {
-			util.Log.Info("Validation finished without errors")
-		} else {
-			util.Log.Info("Deployment finished without errors")
-		}
-		err = deleteConfigs(apis, environments, path, dryRun, fileReader)
-		if err != nil {
-			util.Log.Error("Deletion of configs failed with error %s\n", err)
-			statusCode = -1
-		}
+		log.Fatal(err)
+		return 1
 	}
 
-	return statusCode
+	return 0
 }
 
-func parseInputCommand(args []string, fileReader util.FileReader) (dryRun bool, verbose bool, environments map[string]environment.Environment, project string, path string, errorList []error, flagError error) {
-
-	// define flags
-	var environmentsFile string
-	var specificEnvironment string
-	var versionFlag bool
+func buildCli(fileReader util.FileReader) *cli.App {
+	app := cli.NewApp()
```

Comment:
Please add `app.EnableBashCompletion = true` to enable the bash completion.

https://github.com/urfave/cli/blob/master/docs/v2/manual.md#bash-completion

Reply:
I didn't include it by default, as this needs some more ritual to get working:
https://github.com/urfave/cli/blob/master/docs/v2/manual.md#enabling

And I guess this will break by us switching the cli based on an env variable, 
as most people will use it like `NEW_CLI=1 ./monaco` (at least I assume this).

Reply:
But I can activate it if needed :+1:

Reply:
Hmm, I should have read more than the first paragraph :smile:. It, would make sense to activate/document the completion in its own ticket. Maybe there is a way to implement it with some graceful handling.

Reply:
I guess a new ticket is the way to go, as there are too many unanswered questions. 

We could provide the completions for the old cli and once it is deprecated switch over
to the new. 

But I guess this would required two user interactions.

Another way would be to serve the completions via a command like `monaco completions bash`.
The user the would put something like `source $(monaco compeltions bash)` into their .bashrc`
and it would automatically switch over to the new completions once we update.